src/tests/regression/reported/issue-552-send_by_ref-singleton.f90

```diff
@@ -0,0 +1,33 @@
+program main
+   implicit none
+
+   type co_obj ! the test does not fail with pure coarrays, only when using derived types
+      real, allocatable :: a(:, :, :)[:], b(:, :, :)
+   end type
+
+   integer :: me, nimg
+   type(co_obj) :: co
+
+   me = this_image()
+   nimg = num_images()
+
+   if (nimg /= 2) error stop 1
+
+   allocate(co % a(1, 10, 20)[*], co % b(1, 10, 20))
+   call random_number(co % b)
+
```

Comment:
While we're not actually checking the output anywhere, don't we need a synchronization here, so that a is assigned from b *after* `random_number` returns?

Reply:
@zbeekman 
Will include your changes ASAP !

Reply:
Amazing, cheers!